src/main/java/ui/Parser.java

```diff
@@ -18,50 +18,50 @@ public static String getCommandWord(String fullCommand) {
         return command;
     }
 
-    public static Todo createTodo(String fullCommand) {
+    public static Todo createTodo(String fullCommand) throws DukeException {
         String description = fullCommand.substring("todo".length()).trim();
-        assert (("todo " + description).equals(fullCommand)) : "Something went wrong during the substring for todo description";
-        if (description.isEmpty()) {
-            System.out.println("Empty description for TODO");
+        //Use isBlank() rather than isEmpty(). isBlank() checks for whitespace
+        if (description.isBlank()) {
+            throw new DukeException("Empty description for TODO");
         } else {
             return new Todo(description);
         }
-        return new Todo(description);
+
     }
 
-    public static Deadline createDeadLine(String fullCommand) {
+    public static Deadline createDeadLine(String fullCommand) throws DukeException {
         int idxOfBy = fullCommand.indexOf("/by");
         if (idxOfBy < 0) {
-            System.out.println("Deadline does not contain /by");
+            throw new DukeException("Deadline does not contain /by");
         }
 
         String description = fullCommand.substring(0, idxOfBy).substring("deadline".length()).trim();
-        assert (fullCommand.contains(description)) : "Something went wrong during the substring for deadline description";
+        //Use isBlank() rather than isEmpty(). isBlank() checks for whitespace
         if (description.isEmpty() || description.equals("")) {
-            System.out.println("Empty description for DEADLINE");
+            throw new DukeException("Empty description for DEADLINE");
         }
         String deadline = fullCommand.substring(idxOfBy, fullCommand.length()).substring("/by".length()).trim();
         if (deadline.isEmpty() || deadline.equals("")) {
```

Comment:
I never know isEmpty command. I always checked if the string size =0. Thanks for the tips!